lib/modules/manager/gomod/artifacts.ts

```diff
@@ -275,23 +276,38 @@ export async function updateArtifacts({
         config.postUpdateOptions?.includes('gomodTidy1.17') === true ||
         config.postUpdateOptions?.includes('gomodTidyE') === true ||
         (config.updateType === 'major' && isImportPathUpdateRequired));
+
     if (isGoModTidyRequired) {
       args = 'mod tidy' + tidyOpts;
       logger.debug('go mod tidy command included');
       execCommands.push(`${cmd} ${args}`);
     }
 
+    const goWorkSumFileName = upath.join(goModDir, 'go.work.sum');
     if (useVendor) {
-      args = 'mod vendor';
-      logger.debug('go mod tidy command included');
-      execCommands.push(`${cmd} ${args}`);
-      if (isGoModTidyRequired) {
-        args = 'mod tidy' + tidyOpts;
-        logger.debug('go mod tidy command included');
+      // If go env GOWORK returns a non-empty path, check that it exists and if
+      // it does, then use go workspace vendoring.
+      const goWorkEnv = await exec(`${cmd} env GOWORK`, execOptions);
+      const goWorkFile = goWorkEnv?.stdout?.trim() || '';
+      const useGoWork = goWorkFile.length && ((await readLocalFile(goWorkFile)) !== null);
+      if (useGoWork) {
+        logger.debug('No go.work found');
+      }
+
+      if (useGoWork) {
+        args = 'work vendor';
+        logger.debug('using go work vendor');
```

Comment:
use trace log level

Reply:
The existing log messages and pretty much everything else is using debug, using trace here would be inconsistent, are you sure you want trace here? What about the rest of the logs, or existing logs?

Reply:
Also; these debug logs are really useful for debugging what it's doing, idk how to get trace logs but debug logs here are useful for end-users troubleshooting.

Reply:
you can remap log level at runtime, current logging is too verbose, we should lower almost all messages

https://docs.renovatebot.com/configuration-options/#loglevelremap

Reply:
I'm ok to include this at debug level, particularly considering we have existing statements like `logger.debug('go mod tidy command included');`